internal/controller/gateway.go

```diff
@@ -31,46 +31,107 @@ import (
 )
 
 type gatewayReconciler struct {
+	ctx          context.Context
 	client       client.Client
 	eventHandler cache.ResourceEventHandler
-	logrus.FieldLogger
+	log          logrus.FieldLogger
+	// controllerName is the configured controller of managed gatewayclasses.
+	controllerName string
+	// referencedClass is the gatewayclass referenced by managed gateways.
+	referencedClass *gatewayapi_v1alpha1.GatewayClass
```

Comment:
It looks like this is set but never used - can it be dropped?

Reply:
👍